content/blog/excluding-targets-from-stack-operations/index.md

```diff
@@ -0,0 +1,265 @@
+
+{{% chooser language "javascript,typescript,python,go,csharp" %}}
+
+{{% choosable language javascript%}}
+
+```javascript
+...
+
+// A stub resource to group all our drafts
+const drafts = new pulumi.ComponentResource('ComponentResource', 'drafts')
+
+for (const file in await glob('drafts/**/*.html')) {
+  new aws.s3.BucketObject(`draft-${file}`, {
+    source: new pulumi.asset.FileAsset(`drafts/${file}`),
+    ...
+  }, { parent: drafts })
+}
+
+...
+```
+
+{{% /choosable %}}
+
+{{% choosable language typescript%}}
+
+```typescript
+...
+
+// A stub resource to group all our drafts
+const drafts: pulumi.ComponentResource =
+  new pulumi.ComponentResource('ComponentResource', 'drafts')
+
+for (const file in await glob('drafts/**/*.html')) {
+  new aws.s3.BucketObject(`draft-${file}`, {
+    source: new pulumi.asset.FileAsset(`drafts/${file}`),
+    ...
+  }, { parent: drafts })
+}
+
+...
+```
+
+{{% /choosable %}}
+
+{{% choosable language python %}}
+
+```py
+...
+
+# A stub resource to group all our drafts
+drafts = pulumi.ComponentResource(t="ComponentResource", name="drafts")
+
+for file_path in glob("posts/**/*.html"):
+    aws.s3.BucketObject(file_path,
+        source = pulumi.FileAsset(file_path),
+	 opts = pulumi.ResourceOptions(parent = drafts),
+        ....
+    )
+
+...
+```
+
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+drafts := &DraftGroupComponent{}
+err = ctx.RegisterComponentResource("ComponentResource", "drafts", drafts)
+if err != nil { return err }
+
+...
+
+files, err := filepath.Glob("posts/*.html")
+if err != nil { return err }
+
+for _, file := range files {
+  _, err := s3.NewBucketObject(ctx, file, &s3.BucketObjectArgs{
+    Key: pulumi.String(file),
+    ...
+}, pulumi.Parent(drafts))
+
+  if err != nil { return err }
+}
+
+...
+```
+
+{{% /choosable %}}
+
+{{% choosable language csharp %}}
+
+```csharp
+public class MyComponentResource : ComponentResource { ... }
+
+...
+
+var files = Directory.GetFiles("posts", "*.html");
+var drafts = new MyComponentResource("drafts");
+
+foreach (var file in files)
+{
+    var bucketObject = new BucketObjectv2(file, new BucketObjectv2Args
+    {
+        Source = new FileAsset(file),
+        ...
+    }, new CustomResourceOptions
+    {
+        Parent = drafts
+    });
+}
+
+...
+```
+
+{{% /choosable %}}
+
+{{% /chooser %}}
+
+In this setup, we now have a parent resource for all drafts. Using `--exclude-dependents`, we can now exclude everything under this parent resource without having to enumerate all of them individually:
+
+```shell
+pulumi up --exclude <URN>::ComponentResource::drafts --exclude-dependents
+```
+
+This command will exclude all drafts from the `up` operation, regardless of how many we have or how they’re named. We now have a nice, scalable way to manage our drafts across production and development environments\!
```

Comment:
Does the backslash need escaping?

Reply:
Probably not - I exported this bit from Google Docs, so it might just be being overly cautious!